crates/adapters/src/integrated/postgres/output.rs

```diff
@@ -164,6 +175,81 @@ impl Drop for PostgresOutputEndpoint {
 
 const PG_CONNECTION_VALIDITY_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(10);
 
+fn set_certs(builder: &mut SslConnectorBuilder, config: &PostgresWriterConfig) -> AnyResult<()> {
+    let Some(ca_cert) = &config.ssl_ca_pem else {
+        return Ok(());
+    };
+
+    let ca_cert_path = {
+        let mut file =
+            NamedTempFile::new().context("failed to create tempfile to write CA certificate to")?;
+        let path = file.path().to_owned();
+
+        file.write_all(ca_cert.as_bytes())
+            .context("failed to write CA certificate to tempfile")?;
+        file.flush()
+            .context("failed to flush CA certificate to tempfile")?;
+
+        let (_, path) = file.keep()?;
+        path
+    };
+
+    builder
+        .set_ca_file(ca_cert_path)
+        .context("failed to set CA certificate in SSL connector")?;
+
+    if let (Some(client_cert), Some(client_key)) = (&config.ssl_client_pem, &config.ssl_client_key)
+    {
+        tracing::info!("postgres: using client certificate and key to connect to postgres");
+        let cert = X509::from_pem(client_cert.as_bytes())
+            .context("failed to parse client certificate as X509")?;
+        builder
+            .set_certificate(&cert)
+            .context("failed to set client certificate in SSL connector")?;
+        let rsa = Rsa::private_key_from_pem(client_key.as_bytes())
+            .context("failed to parse client private key as RSA")?;
+        let key = PKey::from_rsa(rsa).context("failed to client private key from RSA")?;
+        builder
+            .set_private_key(key.as_ref())
+            .context("failed to set client private key")?;
+    };
+
+    Ok(())
+}
+
+fn connect(config: &PostgresWriterConfig) -> Result<Client, BackoffError> {
+    let connector_config = config.clone();
+
+    let pgcnf = postgres::Config::from_str(&config.uri).map_err(|e| {
+        BackoffError::Permanent(anyhow!("error parsing postgres connection string: {e}"))
+    })?;
+
+    let client = if let Some(ca_cert) = &config.ssl_ca_pem {
+        let mut builder = SslConnector::builder(SslMethod::tls())
+            .map_err(|e| BackoffError::Permanent(anyhow!("failed to build SSL connection: {e}")))?;
+
+        set_certs(&mut builder, config).map_err(BackoffError::Permanent)?;
+
+        let mut connector = MakeTlsConnector::new(builder.build());
+
+        if Some(false) == config.verify_hostname {
+            connector.set_callback(|ctx, _| {
+                tracing::info!("postgres: ssl: disabling hostname verification");
```

Comment:
I think this should be a warning

Reply:
"Disabled hostname verification for the postgresql connector in $where-they-disabled-this. $what-are-the-implicatoins-here"

Reply:
this print happens twice(?) at the end of the PR there is a very similar line like this

Reply:
Yeah, that is to disable hostname verification when connecting to the main DB from pipeline-manager.